src/devices/src/virtio/block/io/async_io.rs

```diff
@@ -197,19 +197,10 @@ impl<T> AsyncFileEngine<T> {
     }
 
     fn drain_submission_queue(&mut self) -> Result<(), Error> {
-        // Drain the submission queue.
-        loop {
-            // In order for this loop to ever end, we must guarantee that there isn't another thread
-            // submitting ops on the io_uring fd.
-            let sq_len = self.ring.pending_sqes().map_err(Error::IoUring)?;
-            if sq_len == 0 {
-                break;
-            }
-
-            self.ring.submit_and_wait(sq_len).map_err(Error::IoUring)?;
-        }
-
-        Ok(())
+        self.ring
+            .submit_and_wait_all()
+            .map(|_| ())
+            .map_err(Error::IoUring)
```

Comment:
This function has the same body as `kick_submission_queue_and_wait_all` above, which is used for testing purposes. Why do we need both? Am I missing something?